components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,343 @@
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.error
+import androidx.compose.ui.semantics.onClick
+import androidx.compose.ui.semantics.password
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.stateDescription
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.atom.button.SecondaryButton
+import uk.gov.hmrc.components.compose.atom.text.ErrorText
+import uk.gov.hmrc.components.compose.molecule.input.PasswordInput.ANIMATION_DURATION
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+
+@Composable
+fun PasswordInputView(
+    modifier: Modifier = Modifier,
+    passwordTrailingButton: PasswordTrailingButton,
+    value: String? = null,
+    onInputValueChange: ((String) -> Unit)? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    hintText: String? = null,
+    hintContentDescription: String? = null,
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    numericOnly: Boolean = true,
+    maxChars: Int? = null,
+    requiredSequencesSpacing: Boolean = true,
+) {
+    // pattern matches a non decimal number
+    val nonDecimalPattern = remember { Regex("^([0-9]*)$") }
+
+    fun decimalPatternChecker(input: String, localValue: String) = when {
+        numericOnly && input.matches(nonDecimalPattern) -> input
+        !numericOnly -> input
+        else -> localValue
+    }
+
+    PasswordTextInputView(
+        modifier = modifier.semantics { password() },
+        value = value,
+        onInputValueChange = onInputValueChange,
+        inputFilter = { it: String, localValue: String ->
+            decimalPatternChecker(
+                it,
+                localValue
+            )
+        },
+        labelText = labelText,
+        labelContentDescription = labelContentDescription,
+        hintText = hintText,
+        hintContentDescription = hintContentDescription,
+        errorText = errorText,
+        errorContentDescription = errorContentDescription,
+        passwordTrailingButton = PasswordTrailingButton(
+            showButtonText = passwordTrailingButton.showButtonText,
+            showButtonContentDescription = passwordTrailingButton.showButtonContentDescription,
+            showButtonStateDescription = passwordTrailingButton.showButtonStateDescription,
+            hideButtonText = passwordTrailingButton.hideButtonText,
+            hideButtonContentDescription = passwordTrailingButton.hideButtonContentDescription,
+            hideButtonStateDescription = passwordTrailingButton.hideButtonStateDescription
+        ),
```

Comment:
Do we need to do this? Could we just pass the `passwordTrailingButton` through? Looks like they're the same object type already, so not sure we need to map it to a new object?